tap_hubspot/hubspot.py

```diff
@@ -173,9 +173,27 @@ def streams(self, start_date: datetime, end_date: datetime):
             yield from self.get_submissions()
         elif self.tap_stream_id == "contacts_events":
             yield from self.get_contacts_events()
+        elif self.tap_stream_id == "deal_properties":
+            yield from self.get_properties("deals")
+        elif self.tap_stream_id == "contact_properties":
+            yield from self.get_properties("contacts")
+        elif self.tap_stream_id == "company_properties":
+            yield from self.get_properties("companies")
         else:
             raise NotImplementedError(f"unknown stream_id: {self.tap_stream_id}")
 
+    def get_properties(self, object_type: str):
+        path = f"/crm/v3/properties/{object_type}"
+        data_field = "results"
+        replication_path = ["updatedAt"]
+        offset_key = "after"
+        yield from self.get_records(
+            path,
+            replication_path,
+            data_field=data_field,
+            offset_key=offset_key,
+        )
```

Comment:
I don't know if I prefer this to the (seemingly) established pattern of having a dedicated method per stream, but I guess properties are more of a generic thing so it makes sense?